src/awesome/client.rs

```diff
@@ -42,9 +43,60 @@ impl<'lua> ToLua<'lua> for Client<'lua> {
     }
 }
 
-impl UserData for ClientState {}
+impl UserData for ClientState {
+    fn add_methods(methods: &mut UserDataMethods<Self>) {
+        methods.add_method("geometry", |lua, state, (geometry,): (Value,)| {
+            match geometry {
+                Value::Table(table) => {
+                    let area = Area {
+                        origin: Origin { x: table.get("x")?, y: table.get("y")? },
+                        size: Size { width: table.get("width")?, height: table.get("height")? }
+                    };
+
+                    state.view.move_resize(area);
+
+                    Ok(Value::Nil.to_lua(lua))
+                },
+                _ => {
+                    let table = lua.create_table()?;
+                    let geometry = state.view.geometry();
+                    let Origin { x, y } = *state.view.origin.lock().unwrap();
+                    table.set("x", geometry.origin.x + x)?;
+                    table.set("y", geometry.origin.y + y)?;
+                    table.set("width", geometry.size.width)?;
+                    table.set("height", geometry.size.height)?;
+                    Ok(table.to_lua(lua))
+                }
+            }
+        });
+
+        methods.add_meta_function(MetaMethod::Index, |lua, (class, index): (ClientState, Value)| {
```

Comment:
For this method you can use the same API as the others as well and just [use the special `"__index"` name](https://github.com/way-cooler/way-cooler/blob/master/src/awesome/screen.rs#L144).

Reply:
This api also only works only for the global.